TempleEight/docs/guide/value-constraints.md

```diff
@@ -0,0 +1,75 @@
+---
+id: value-constraints
+title: Value Constraints
+sidebar_label: Value Constraints
+---
+
+Sometimes you need more control over the values that are acceptable in your data model, for these cases Temple provides a suite of Value Constraints:
+extra metadata you can attach to properties that allow values to be restricted in some way.
+
+## Primitive Restrictions
+
+Many of Temple's primitive data types support optional parameters when you declare them, to give bounds on their values.
+
+An example of this is when an integer is defined. We'll take the `ExampleService` from the [Getting Started](getting-started) guide:
+
+```
+ExampleService: service {
+  foo: string;
+  bar: int;
+}
+```
+
+If for example we wanted to constrain the `bar` property to be only positive numbers, we could write:
+
+```
+ExampleService: service {
+  foo: string;
+  bar: int(0);
```

Comment:
Might be worth adding the argument name here, then saying it can be dropped?

Reply:
I forgot we still supported those!